cheatsheets/Authentication_Cheat_Sheet.md

```diff
@@ -126,32 +126,32 @@ Example using pseudo-code for a login feature:
 *First implementation using the "quick exit" approach*
 
 ```
-IF USER_EXISTS(login) THEN
+IF USER_EXISTS(username) THEN
     password_hash=HASH(password)
-    IS_VALID=LOOKUP_CREDENTIALS_IN_STORE(login, password_hash)
+    IS_VALID=LOOKUP_CREDENTIALS_IN_STORE(username, password_hash)
     IF NOT IS_VALID THEN
-        RETURN Error("Invalid User or Password!")    
+        RETURN Error("Invalid Username or Password!")    
     ENDIF
 ELSE
-   RETURN Error("Invalid User or Password!")
+   RETURN Error("Invalid Username or Password!")
 ENDIF
 ```
 
-With this implementation: We can have a time-based discrepancy factor because, if the user exists then two time-taking operations are performed otherwise only one if the user does not exist.
+It can be clearly seen that if the user doesn't exist, the application will directly throw out an error. Otherwise, when the user exists and the password doesn't, it is apparent that there will be more processing before the application errors out. In return, the response time will be different for the same error, allowing for the attacker to differentiate between a wrong username and a wrong password.
```

Comment:
`allowing the attacker to` instead of `allowing for the attacker to`. Apologies 😄

Reply:
No problem 😄 
Fix applied in coming commit.